Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/roles_field.tsx

```diff
@@ -17,34 +16,30 @@ interface Properties {
 
 /** Displays a panel of icons highlighting an account's roles. */
 export class RolesField extends React.Component<Properties, {}> {
-  constructor(props: Properties) {
-    super(props);
-  }
 
   public render(): JSX.Element {
     return (
       <HBoxLayout width={RolesField.COMPONENT_WIDTH}
           height={RolesField.IMAGE_SIZE}>
-        <RolesIcon icon = {Nexus.AccountRoles.Role.TRADER}
-          isSet = {this.props.roles.test(Nexus.AccountRoles.Role.TRADER)}
+        <RoleIcon role={Nexus.AccountRoles.Role.TRADER}
```

Comment:
Factor all of this out by defining a local component directly inside of the ```render``` method, something along the lines of:
```
public render(): JSX.Element {
  const Icon = (props: any) => {
    return (<RoleIcon role={props.role} isSet={this.props.roles.test(props.role)}
      onClick={() => this.props.onClick(props.role)}/>);
  }
  ...
  <Icon role={Nexus.AccountRoles.Role.TRADER}/>
  <Padding...>
  <Icon role={Nexus.AccountRoles.Role.MANAGER}/>
   ...
}
```